src/main/java/org/dependencytrack/util/NotificationUtil.java

```diff
@@ -347,4 +352,55 @@ public static String generateNotificationTitle(String messageType, Project proje
     private static void sendNotificationToKafka(UUID projectUuid, Notification notification){
         new KafkaEventDispatcher().dispatchAsync(projectUuid, notification);
     }
+    public static void sendNotificationForCompletedVulnerabilityScan(VulnerabilityScan vulnscan) {
+        try (QueryManager qm = new QueryManager()) {
+            Project project = qm.getObjectByUuid(Project.class, vulnscan.getTargetIdentifier());
+            List<Component> componentList = qm.getAllComponents(project);
```

Comment:
This could become quite inefficient for large projects (>= 1000 components).

Assuming the majority of components will not be vulnerable, we're loading a ton of data here that we don't really need. And for each one of those components, we're also executing a `getAllVulnerabilities` query.

It might make sense to use `getFindings` here instead:

https://github.com/DependencyTrack/hyades-apiserver/blob/a1ae09e5d950123fffc5719b7c515b33864b3bb2/src/main/java/org/dependencytrack/persistence/FindingsQueryManager.java#L260-L267C4

There you already get Component-Vulnerability pairs, and it only considers components that are in fact vulnerable.